### PR TITLE
fix(core): allow git URLs not ending in .git

### LINKED ISSUE #1356: Garden fails to source a git repo that doesn't end in ".git"
## Bug
Garden uses `is-git-url` (https://github.com/jonschlinkert/is-git-url) to validate a git URL. Unfortunately this doesn't allow for git URLs that don't end in `.git` (something that Azure DevOps does for example:     `git@ssh.dev.azure.com:v3/organization/project/repo`).

### Current Behavior
Validating a garden module that includes source URLs not ending in `.git` fails.
```
$ garden build
Build 

Error validating project (/garden.yml): key .sources[1][repositoryUrl] must be a valid Git repository URL, key .sources[1][repositoryUrl] must be a valid uri with a scheme matching the file pattern
```

### Expected behavior
Validating and checking out the git URL succeeds.

### Reproducible example
Create a directory with this garden.yaml file:
```
kind: Project
name: git-test
environments:
  - name: local
    providers:
      - name: local-kubernetes
        namespace: "git-test"
sources:
  - name: fake-git-url
    repositoryUrl: git@ssh.dev.azure.com:v3/organization/project/repo#master
```
Run `garden build`

### Your environment
`garden version`
0.10.15

`kubectl version`
Client Version: version.Info{Major:"1", Minor:"16", GitVersion:"v1.16.3", GitCommit:"b3cbbae08ec52a7fc73d334838e18d17e8512749", GitTreeState:"clean", BuildDate:"2019-11-18T14:56:51Z", GoVersion:"go1.12.13", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.8", GitCommit:"1da9875156ba0ad48e7d09a5d00e41489507f592", GitTreeState:"clean", BuildDate:"2019-11-14T05:19:20Z", GoVersion:"go1.12.10", Compiler:"gc", Platform:"linux/amd64"}

`docker version`
Client:
 Version:           18.09.7
 API version:       1.39
 Go version:        go1.10.1
 Git commit:        2d0083d
 Built:             Fri Aug 16 14:20:06 2019
 OS/Arch:           linux/amd64
 Experimental:      false

Server:
 Engine:
  Version:          18.09.9
  API version:      1.39 (minimum version 1.12)
  Go version:       go1.13.4
  Git commit:       9552f2b
  Built:            Sat Nov 16 01:07:48 2019
  OS/Arch:          linux/amd64
  Experimental:     false

### PR DESCRIPTION
**Pull request by thsig**
Fixes #1356.

### CODE REVIEW

**Comment by edvald** [comment]
Good catch. Perhaps we should add tests for things we do _not_ want to match.

File: garden-service/src/config/common.ts [hunk 1]
```diff
@@ -226,6 +225,7 @@ export const absolutePathRegex = /^\/.*/ // Note: Only checks for the leading sl
 export const identifierRegex = /^(?![0-9]+$)(?!.*-$)(?!-)[a-z0-9-]{1,63}$/
 export const userIdentifierRegex = /^(?!garden)(?=.{1,63}$)[a-z][a-z0-9]*(-[a-z0-9]+)*$/
 export const envVarRegex = /^(?!garden)[a-z_][a-z0-9_\.]*$/i
+export const gitUrlRegex = /(?:git|ssh|https?|git@[-\w.]+):(\/\/)?(.*?)(\/?|\#[-\d\w._]+?)$/
```
Comment (dcharbonnier) [inline garden-service/src/config/common.ts]: should support `/` in the branch name
`(?:git|ssh|https?|git@[-\w.]+):(\/\/)?(.*?)(\.git)(\/?|\#[-\d\w._\/]+?)$`

**Comment by edvald** [comment]
@thsig looks good, just needs to be rebased to resolve the conflicts.

**Comment by thsig** [comment]
@edvald: Since we moved back to using our own regex here, would you like to have another look? Or should we just merge once tests pass?

File: garden-service/src/config/common.ts [hunk 2]
```diff
@@ -209,16 +208,16 @@ joi = joi.extend({
...
   },
   validate(value: string, { error }) {
-    // Make sure it's a string
-    const baseSchema = Joi.string()
+    const baseSchema = joi.string().regex(gitUrlRegex)
+    // const baseSchema = Joi.string().uri({
+    //   scheme: ["git", "ssh", "git+ssh", "https", "git+https"],
+    //   // scheme: ["git", "ssh", /git\+ssh/, /https?/, /git\+https?/],
+    //   // scheme: ["git", "ssh", "git+ssh", /https?/, /git\+https?/],
+    // })
```
Comment (edvald) [inline garden-service/src/config/common.ts]: Clean up commented lines

File: garden-service/test/unit/src/config/common.ts [hunk 3]
```diff
@@ -318,6 +325,13 @@ describe("joiRepositoryUrl", () => {
     expect(result.error).to.exist
   })
 
+  it("should reject values missing a schema", () => {
+    const url = "garden-io/garden-example-remote-sources-web-services.git#my-tag"
+    const schema = joi.gitUrl()
```
Comment (edvald) [inline garden-service/test/unit/src/config/common.ts]: Minor thing, but probably best to consistently test against `joiRepositoryUrl` in all the tests

**Pull request merged**